src/Tizen.NUI.WindowSystem/src/public/EdgeDragData.cs

```diff
@@ -0,0 +1,167 @@
+        {
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.GestureGrab(_gesture.GetNativeHandle(), _handle);
+            _gesture.ErrorCodeThrow(res);
+            Log.Debug(LogTag, "EdgeDragData.GestureGrab");
+        }
+
+        /// <summary>
+        /// Ungrabs a global gesture.
+        /// </summary>
+        public void UngrabGesture()
+        {
+            Interop.InputGesture.ErrorCode res = Interop.InputGesture.GestureUngrab(_gesture.GetNativeHandle(), _handle);
+            _gesture.ErrorCodeThrow(res);
+            Log.Debug(LogTag, "EdgeDragData.GestureUngrab");
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        ~EdgeDragData()
+        {
+            Dispose(false);
+            Log.Debug(LogTag, "EdgeDragData Destroyed");
+        }
+
+        /// <summary>
+        /// Dispose.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc/>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposed)
+            {
+                Interop.InputGesture.ErrorCode res = Interop.InputGesture.EdgeDragFree(_gesture.GetNativeHandle(), _handle);
+                _gesture.ErrorCodeThrow(res);
+
+                if (disposing)
+                {
+                    _handle?.Dispose();
+                    _handle = null;
+                }
+            }
+        }
+
+        internal SafeGestureHandle GetNativeHandle()
+        {
+            return _handle;
+        }
+
+        /// <summary>
+        /// Emits the event when the edge drag event comes
+        /// </summary>
+        public event EventHandler<EdgeDragEventArgs> EdgeDragEventHandler
+        {
+            add
+            {
+                if (_handle == null)
+                {
+                    Log.Info(LogTag, "EdgeDragData is NULL.");
+                    return;
+                }
+                _delegate = (IntPtr userData, int mode,  int fingers, int cx, int cy, int edge) =>
+                {
+                    EdgeDragEventArgs args = new EdgeDragEventArgs(mode, fingers, cx, cy, edge);
+                    Log.Info(LogTag, "EdgeDrag Event received. mode: " + mode + ", fingers: " + fingers);
+                    _eventHandler?.Invoke(null, args);
+                };
+                Interop.InputGesture.ErrorCode error = Interop.InputGesture.SetEdgeDragCb(_gesture.GetNativeHandle(), _handle, _delegate, IntPtr.Zero);
+                if (error != Interop.InputGesture.ErrorCode.None)
+                {
+                    Log.Info(LogTag, "error" + error.ToString());
+                }
+                else
+                {
+                    _eventHandler += value;
+                }
+            }
+            remove
+            {
+                _eventHandler -= value;
+            }
+        }
+
+    }
+}
```

Comment:
could you add new line at the end of file?
reference: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline